gammapy/datasets/core.py

```diff
@@ -433,10 +435,15 @@ def stack_reduce(self, name=None, nan_to_num=True):
             raise ValueError(
                 "Stacking impossible: all Datasets contained are not of a unique type."
             )
+        if self[0].tag == "FluxPointsDataset":
+            raise ValueError("Stacking not defined for FluxPointsDataset")
```

Comment:
Wouldn't it be better to move this to a `stack` method for `FluxPointsDataset` and return a specific message there with a `NotImplementedError`?